src/TesApi.Web/Storage/DefaultStorageAccessProvider.cs

```diff
@@ -204,8 +205,16 @@ public override async Task<string> GetInternalTesTaskBlobUrlAsync(TesTask task,
             if (task.Resources?.ContainsBackendParameterValue(TesResources.SupportedBackendParameters
                     .internal_path_prefix) == true)
             {
+                var internalPath = $"{task.Resources.GetBackendParameterValue(TesResources.SupportedBackendParameters.internal_path_prefix).Trim('/')}{normalizedBlobPath}";
+
+                if (storageOptions.ExecutionsContainerName is not null &&
+                    !internalPath.StartsWith(storageOptions.ExecutionsContainerName, StringComparison.OrdinalIgnoreCase))
```

Comment:
Would this be better?
internalPath.StartsWith(storageOptions.ExecutionsContainerName + "/",